### PR TITLE
feat: Support custom endpoint

### PR DESCRIPTION
**Pull request by maiqingqiang**
Resolving the issue of some regions being unable to access https://huggingface.co

### CODE REVIEW

File: Sources/Hub/HubApi.swift [hunk 1]
```diff
@@ -8,21 +8,22 @@
 import Foundation
 
 public struct HubApi {
-    let endpoint = "https://huggingface.co/api"
     var downloadBase: URL
-    var hfToken: String? = nil
+    var hfToken: String?
+    var hfEndpoint: String
```
Comment (pcuenca) [inline Sources/Hub/HubApi.swift:13]: Why not keep `endpoint` here?

File: Sources/Hub/HubApi.swift [hunk 2]
```diff
@@ -8,21 +8,22 @@
...
     
     public typealias RepoType = Hub.RepoType
     public typealias Repo = Hub.Repo
     
-    public init(downloadBase: URL? = nil, hfToken: String? = nil) {
+    public init(downloadBase: URL? = nil, hfToken: String? = nil, hfEndpoint: String = "https://huggingface.co") {
```
Comment (pcuenca) [inline Sources/Hub/HubApi.swift:18]: 👍

File: Sources/Hub/HubApi.swift [hunk 3]
```diff
@@ -243,9 +245,8 @@ public extension Hub {
     }
 }
 
-public extension Array<String> {
+public extension [String] {
```
Comment (pcuenca) [inline Sources/Hub/HubApi.swift:248]: This is unrelated, but ok.

**Review by pcuenca — COMMENTED** [review]
Thanks a lot! This is very useful, I only have one question about changing the property name. If anything, I'd remove `hf` from `hfToken` rather than adding it to `endpoint`.

File: Sources/Hub/HubApi.swift (hunk 3 above)
Comment (maiqingqiang) [reply Sources/Hub/HubApi.swift:248]: This has been processed by the swift-format tool.

File: Sources/Hub/HubApi.swift (hunk 1 above)
Comment (maiqingqiang) [reply Sources/Hub/HubApi.swift:13]: Because the token also has the "hf" prefix.😂

**Comment by maiqingqiang** [comment]
> Thanks a lot! This is very useful, I only have one question about changing the property name. If anything, I'd remove `hf` from `hfToken` rather than adding it to `endpoint`.

I also think using `endpoint` is better. So, should I change `hfToken` to `token` on my end?

File: Sources/Hub/HubApi.swift (hunk 3 above)
Comment (pcuenca) [reply Sources/Hub/HubApi.swift:248]: Yeah, I don't use it. I write what feels clearer in each case, I tend to use the short form for variables, and the spelled out format when working with the type itself, such as in extensions. But it's ok.

**Comment by pcuenca** [comment]
> I also think using `endpoint` is better. So, should I change `hfToken` to `token` on my end?

I'd love that, but `hfToken` is already in the `init` signature, so we could break other people's code if we change it. Coming to think of it, let's keep `endpoint` for the same reason, if that's ok 🙏

**Comment by maiqingqiang** [comment]
> > I also think using `endpoint` is better. So, should I change `hfToken` to `token` on my end?
> 
> I'd love that, but `hfToken` is already in the `init` signature, so we could break other people's code if we change it. Coming to think of it, let's keep `endpoint` for the same reason, if that's ok 🙏

Resolved

File: Sources/Hub/HubApi.swift (hunk 3 above)
Comment (maiqingqiang) [reply Sources/Hub/HubApi.swift:248]: Can we introduce swift-format to unify the code style? Similar to https://github.com/nicklockwood/SwiftFormat?tab=readme-ov-file#git-pre-commit-hook or https://github.com/apple/swift-format with git pre commit hook

**Review by pcuenca — APPROVED** [review]
Thanks a lot!

**Pull request merged**